tools/deps_info.py

```diff
@@ -198,8 +198,18 @@
 
 
 def get_deps_info():
-  if not settings.EXCEPTION_HANDLING and not settings.DISABLE_EXCEPTION_CATCHING:
+  if not settings.EXCEPTION_HANDLING and settings.LINK_AS_CXX:
     _deps_info['__cxa_begin_catch'] = ['__cxa_is_pointer_type']
+    _deps_info['__cxa_find_matching_catch'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_1'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_2'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_3'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_4'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_5'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_6'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_7'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_8'] = ['__cxa_can_catch']
+    _deps_info['__cxa_find_matching_catch_9'] = ['__cxa_can_catch']
```

Comment:
can write a loop here? but, why is this needed in this PR?

Reply:
When exception catching is enabled we force the inclusion of these symbols in emcc.py.   However when exception catchi is disabled at link time we still want to support the used of these functions become some object might have been builld with exception catching enabled.   

Rather then just always include these functions I decided to go back to old way to include these dependencies prior to #13528.   This method works for the non-LTO builds.. LTO build will be handled by the forces dependency `emcc.py`.